Integrate Callout with Graph Investigation to render empty states

albertoblaz opened the pull request:
## Summary

Closes:
- https://github.com/elastic/security-team/issues/14479

Follows-up:
- https://github.com/elastic/kibana/pull/244498

Integrates `<Callout>` component into `<GraphInvestigation>` to render empty states. 

> [!NOTE]
> Users can dismiss callout by clicking on the "X" button, but such action is not persisted as of yet. It will be part of a follow-up ticket.

## How to test - locally in Kibana

Run this command to load archives:

```bash
node scripts/es_archiver load x-pack/solutions/security/test/cloud_security_posture_functional/es_archives/logs_gcp_audit --es-url http://elastic:changeme@localhost:9200 --kibana-url http://elastic:changeme@localhost:5601
node scripts/es_archiver load x-pack/solutions/security/test/cloud_security_posture_functional/es_archives/security_alerts_modified_mappings --es-url http://elastic:changeme@localhost:9200 --kibana-url http://elastic:changeme@localhost:5601
```

### Test Case 1 - Missing all requirements

1. Go to Alerts page and open the graph from the first alert
2. Check callout shows the same message as the screenshot below.

<details><summary>Screenshot</summary>
<img width="1784" height="1562" alt="Screenshot 2025-12-11 at 16 35 57" src="https://github.com/user-attachments/assets/6550c32f-38cb-4906-a1f4-79d69a914b41" />
</details>

### Test Case 2 - Missing Cloud Asset Discovery integration

3. Go to Entity Store page and toggle ON the switch at the top-right corner
4. Go to Alerts page and open the graph from the first alert
5. Check callout shows the same message as the screenshot below.

<details><summary>Screenshot</summary>
<img width="1786" height="1563" alt="Screenshot 2025-12-11 at 16 35 23" src="https://github.com/user-attachments/assets/7eb91958-b5c3-4885-a165-c1f89aaaaab8" />
</details>

<details><summary>Link opens "Cloud Asset Discovery" integration page</summary>
<img width="1508" height="754" alt="Screenshot 2025-12-15 at 11 37 02" src="https://github.com/user-attachments/assets/a730a5c7-76f9-457c-b272-27075fbbe056" />
</details>

### Test Case 3 - Missing Entity Store

6. Go to Entity Store page and toggle OFF the switch at the top-right corner
7. Go to Alerts page and open the graph from the first alert
8. Check callout shows the same message as the screenshot below.

<details><summary>Screenshot</summary>
<img width="1789" height="1568" alt="Screenshot 2025-12-11 at 16 32 37" src="https://github.com/user-attachments/assets/4d54dfb7-c458-4547-9e2a-6383d26f7a90" />
</details>

<details><summary>Link opens Entity Store page</summary>
<img width="2509" height="1658" alt="Screenshot 2025-12-11 at 18 08 35" src="https://github.com/user-attachments/assets/72edf234-1068-45e4-8514-471b1a006218" />
</details>

### Test Case 4 - Entity not available in Entity Store

9. Revert commit in this branch named `'Revert "DEBUG - availableEntityStore: false"'`
10. Go to Alerts page and open the graph from the first alert
11. Check callout shows the same message as the screenshot below.

<details><summary>Screenshot</summary>
<img width="1784" height="1563" alt="Screenshot 2025-12-11 at 16 30 59" src="https://github.com/user-attachments/assets/0f323711-7098-4873-a167-d6aafb84a4e1" />
</details>

<details><summary>Link opens Discover page</summary>
<img width="2512" height="1654" alt="Screenshot 2025-12-11 at 18 12 28" src="https://github.com/user-attachments/assets/499eb2fe-1189-4cb8-842f-4bf63b913cfd" />
</details>

### Test Case 5 - Unknown entity type

12. Revert last commit to restore changes
13. Go to Alerts page and open the graph from the first alert
14. Check callout shows the same message as the screenshot below.

<details><summary>Screenshot</summary>
<img width="1780" height="1563" alt="Screenshot 2025-12-11 at 16 29 49" src="https://github.com/user-attachments/assets/d7ecb85b-19fc-47ea-b254-6d9f4ef4110f" />
</details>

<details><summary>Link opens Discover page</summary>
<img width="2512" height="1654" alt="Screenshot 2025-12-11 at 18 12 28" src="https://github.com/user-attachments/assets/499eb2fe-1189-4cb8-842f-4bf63b913cfd" />
</details>

### Test Case 6 - No callout visible

15. Go to Alerts page and open the graph from the first alert
16. Revert `'Revert "DEBUG - entity type & sub_type"`
17. Check callout is hidden because all conditions are met, as in the screenshot below.

<details><summary>Screenshot</summary>
<img width="1791" height="1563" alt="Screenshot 2025-12-11 at 16 27 53" src="https://github.com/user-attachments/assets/10bf362a-f5a6-49ec-b4ea-d46f8591b23f" />
</details>

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

No risk. Feature is gated behind advanced setting.

albertoblaz committed:
Integrate callout into GraphInvestigation

albertoblaz committed:
DEBUG - availableInEntityStore: false

albertoblaz committed:
Revert "DEBUG - availableInEntityStore: false"

This reverts commit 26993534cdec0a494d63fd0d60d257a8dac75b0a.

albertoblaz committed:
DEBUG - type & sub_type

albertoblaz committed:
Revert "DEBUG - type & sub_type"

This reverts commit d57072b2a73b59cb3c8d53ccda8ca0e352996e6f.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-entity-store-performance-from-pr --> Click to trigger **kibana-entity-store-performance-from-pr** for this PR!

albertoblaz committed:
Restore callout.stories.tsx to original structure

albertoblaz committed:
Remove individual hooks from package index.ts

albertoblaz committed:
Merge branch 'main' into graph-show-callout

albertoblaz committed:
Remove for real individual hooks from package index.ts

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/hooks/use_callout_status.ts:
```diff
@@ -0,0 +1,194 @@
+
+// Constants
+const CLOUD_ASSET_DISCOVERY_INTEGRATION_PACKAGE_NAME = 'cloud_asset_inventory' as const;
+const ENTITY_ANALYTICS_API_VERSION = '2023-10-31' as const;
```
Is there a constant I can reuse and import into the `kbn-cloud-security-posture/graph` package instead of  hard-coding the API version here?

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/hooks/use_callout_links.ts:
```diff
@@ -0,0 +1,48 @@
+  // Hard-coded path referencing ENTITY_ANALYTICS_ENTITY_STORE_MANAGEMENT_PATH
+  // from x-pack/solutions/security/plugins/security_solution/common/constants.ts
+  const entityStoreUrl = getUrlForApp('security', {
+    path: '/entity_analytics_entity_store',
```
I wanted to reuse the `ENTITY_ANALYTICS_ENTITY_STORE_MANAGEMENT_PATH` constant but the LLM refused to do so, I think because of a circular dependency. Any chance to bypass that?

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

albertoblaz reviewed (commented):
Notes for reviewers down below

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

albertoblaz committed:
fix: resolve TypeScript errors in use_graph_callout.test.ts

Fix "Cannot invoke an object which is possibly 'undefined'" errors by
destructuring onDismiss from result.current within the type guard blocks.
This properly narrows the discriminated union type for TypeScript.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372059)
* Commit: c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-246034)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-246034-c8089ce5cdbc`

### Failed CI Steps
* [FTR Configs #78](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e8f5-4017-8f6f-38a2a78f19e3)
* [FTR Configs #90](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e8fd-4d55-8d0d-98b2450322a2)
* [FTR Configs #102](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e905-4350-833d-1e365ff20818)
* [FTR Configs #107](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e908-4089-bb2b-5cdab6ae3ea1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e905-4350-833d-1e365ff20818) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372059/jobs/019b0ef2-e905-4350-833d-1e365ff20818/artifacts/019b0f16-7578-4d08-b808-2d89a11885ef) FTR Configs #<span></span>102 / Cloud Security Posture Security Alerts Page - Graph visualization expanded flyout - filter by node
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e8f5-4017-8f6f-38a2a78f19e3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372059/jobs/019b0ef2-e8f5-4017-8f6f-38a2a78f19e3/artifacts/019b0f1b-62d5-4125-9e89-6864fb34ea6c) FTR Configs #<span></span>78 / Search solution tests Index management Search solution nav Index list page "before each" hook for "renders search index details with no documents"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372059#019b0ef2-e908-4089-bb2b-5cdab6ae3ea1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372059/jobs/019b0ef2-e908-4089-bb2b-5cdab6ae3ea1/artifacts/019b0f18-54b0-4c82-b00b-c314b74d22c2) FTR Configs #<span></span>107 / Stateful Observability - Deployment-agnostic APM API integration tests APM alerts error count threshold alert create rule without kql filter produces one alert for each of the opbeans-java and opbeans-php

### History
* :broken_heart: [Build #372022](https://buildkite.com/elastic/kibana-pull-request/builds/372022) failed e78a797a2165c810527da3ed2d7cbcd07e033a34
* :broken_heart: [Build #372016](https://buildkite.com/elastic/kibana-pull-request/builds/372016) failed d053a357b9f486a6d07883e58e0791e74634f83f

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372059","number":372059,"commit":"c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372022","number":372022,"commit":"e78a797a2165c810527da3ed2d7cbcd07e033a34"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372016","number":372016,"commit":"d053a357b9f486a6d07883e58e0791e74634f83f"}],"number":372059}
buildkite-pr-comment-->

albertoblaz committed:
Fix FTR tests, assert visible callout

albertoblaz committed:
Merge branch 'main' into graph-show-callout

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/372280)
* Commit: 09ef49f420093871900d1ca8c104946c38ed3941
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-246034)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-246034-09ef49f42009`

### Failed CI Steps
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/372280#019b121d-8054-4fe8-b02b-4b715d746bf8)
* [FTR Configs #91](https://buildkite.com/elastic/kibana-pull-request/builds/372280#019b121d-8098-45f0-8c2b-564bab92ec83)
* [Build Project Image](https://buildkite.com/elastic/kibana-pull-request/builds/372280#019b11f6-c46d-4d04-a28c-4f78aca02b8d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/372280#019b121d-8054-4fe8-b02b-4b715d746bf8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/372280/jobs/019b121d-8054-4fe8-b02b-4b715d746bf8/artifacts/019b1241-6edf-4a0e-ba75-7166da64f643) FTR Configs #<span></span>11 / alerting api integration security and spaces enabled - Group 4 Alerts alerts getGlobalExecutionKpi should return KPI only from the current space

### History
* :broken_heart: [Build #372059](https://buildkite.com/elastic/kibana-pull-request/builds/372059) failed c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52
* :broken_heart: [Build #372022](https://buildkite.com/elastic/kibana-pull-request/builds/372022) failed e78a797a2165c810527da3ed2d7cbcd07e033a34
* :broken_heart: [Build #372016](https://buildkite.com/elastic/kibana-pull-request/builds/372016) failed d053a357b9f486a6d07883e58e0791e74634f83f

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372280","number":372280,"commit":"09ef49f420093871900d1ca8c104946c38ed3941"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372059","number":372059,"commit":"c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372022","number":372022,"commit":"e78a797a2165c810527da3ed2d7cbcd07e033a34"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372016","number":372016,"commit":"d053a357b9f486a6d07883e58e0791e74634f83f"}],"number":372280}
buildkite-pr-comment-->

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/hooks/use_callout_links.ts:
```diff
@@ -0,0 +1,48 @@
+  }
+
+  // Build URLs using Kibana's getUrlForApp
+  const integrationUrl = `${getUrlForApp(INTEGRATIONS_PLUGIN_ID)}/browse/security/asset_inventory`;
```
you could omit the version number and it will default to the latest version:

https://github.com/user-attachments/assets/5bf40657-3a09-4b97-8945-9b891200e7ed

albertoblaz replied:
Ah cool then I'll update the URL

albertoblaz replied:
@alexreal1314 Updated. Thanks!

elasticmachine committed:
Merge branch 'main' into graph-show-callout

albertoblaz committed:
Update link to integration page

albertoblaz committed:
Check missing entity name

albertoblaz committed:
Merge branch 'main' into graph-show-callout

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/hooks/use_callout_status.ts:
```diff
@@ -0,0 +1,194 @@
+    if (!node.documentsData || !Array.isArray(node.documentsData)) {
+      return false;
+    }
+    return node.documentsData.some((doc) => {
```
@alexreal1314 Updated.

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/hooks/use_callout_links.ts:
```diff
@@ -0,0 +1,48 @@
+ *
+ * @returns Object containing integration, entity store, and discover URLs, or undefined if getUrlForApp is unavailable
+ */
+export const useCalloutLinks = ():
```
`CalloutLinks` is only referenced one, so I don't see a problem inlining the interface

albertoblaz committed:
Merge remote-tracking branch 'upstream/main' into graph-show-callout

albertoblaz committed:
Merge branch 'main' into graph-show-callout

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/374026)
* Commit: 1bcec8f3c2c40583fdcedc8ac44e73666839a584

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/374026#019b27f1-f813-414b-9702-b6f5089f0cb5)

### History
* :yellow_heart: [Build #373011](https://buildkite.com/elastic/kibana-pull-request/builds/373011) was flaky f5f6e95525b64cae072fefcde80f80268432d45f
* :yellow_heart: [Build #372280](https://buildkite.com/elastic/kibana-pull-request/builds/372280) was flaky 09ef49f420093871900d1ca8c104946c38ed3941
* :broken_heart: [Build #372059](https://buildkite.com/elastic/kibana-pull-request/builds/372059) failed c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52
* :broken_heart: [Build #372022](https://buildkite.com/elastic/kibana-pull-request/builds/372022) failed e78a797a2165c810527da3ed2d7cbcd07e033a34
* :broken_heart: [Build #372016](https://buildkite.com/elastic/kibana-pull-request/builds/372016) failed d053a357b9f486a6d07883e58e0791e74634f83f

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374026","number":374026,"commit":"1bcec8f3c2c40583fdcedc8ac44e73666839a584"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373011","number":373011,"commit":"f5f6e95525b64cae072fefcde80f80268432d45f"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372280","number":372280,"commit":"09ef49f420093871900d1ca8c104946c38ed3941"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372059","number":372059,"commit":"c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372022","number":372022,"commit":"e78a797a2165c810527da3ed2d7cbcd07e033a34"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372016","number":372016,"commit":"d053a357b9f486a6d07883e58e0791e74634f83f"}],"number":374026}
buildkite-pr-comment-->

albertoblaz committed:
Merge branch 'main' into graph-show-callout

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/374090)
* Commit: 9ff7ead7960261f2da2fc088987f3486e4460bc2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-246034)

### Failed CI Steps
* [Jest Tests #20](https://buildkite.com/elastic/kibana-pull-request/builds/374090#019b283f-b10f-46e7-8716-b1ef3bc400fa)
* [Detection Engine - Security Solution Cypress Tests #5](https://buildkite.com/elastic/kibana-pull-request/builds/374090#019b2824-bec5-4340-bab5-75ab6aeeafa5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/374090#019b2824-bec5-4340-bab5-75ab6aeeafa5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/374090/jobs/019b2824-bec5-4340-bab5-75ab6aeeafa5/artifacts/019b2862-ce03-474d-86ed-0a175b6bca5c) Detection Engine - Security Solution Cypress Tests #<span></span>5 / Rule gaps auto fill status Platinum user flows View gap fill scheduler logs and filter by status View gap fill scheduler logs and filter by status
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/374090#019b283f-b10f-46e7-8716-b1ef3bc400fa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/374090/jobs/019b283f-b10f-46e7-8716-b1ef3bc400fa/artifacts/019b2852-9f7c-4af0-b4e2-fd2ba5fc10d1) Jest Tests #<span></span>20 / useSyntheticsRules dispatches enableDefaultAlertingSilentlyAction when canSave is true and popover opens
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/374090#019b283f-b10f-46e7-8716-b1ef3bc400fa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/374090/jobs/019b283f-b10f-46e7-8716-b1ef3bc400fa/artifacts/019b2852-9f7b-4a5f-b76e-93d5df1e1af6) Jest Tests #<span></span>20 / useSyntheticsRules dispatches getDefaultAlertingAction when canSave is false and popover opens

### History
* :broken_heart: [Build #374026](https://buildkite.com/elastic/kibana-pull-request/builds/374026) failed 1bcec8f3c2c40583fdcedc8ac44e73666839a584
* :yellow_heart: [Build #373011](https://buildkite.com/elastic/kibana-pull-request/builds/373011) was flaky f5f6e95525b64cae072fefcde80f80268432d45f
* :yellow_heart: [Build #372280](https://buildkite.com/elastic/kibana-pull-request/builds/372280) was flaky 09ef49f420093871900d1ca8c104946c38ed3941
* :broken_heart: [Build #372059](https://buildkite.com/elastic/kibana-pull-request/builds/372059) failed c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52
* :broken_heart: [Build #372022](https://buildkite.com/elastic/kibana-pull-request/builds/372022) failed e78a797a2165c810527da3ed2d7cbcd07e033a34

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374090","number":374090,"commit":"9ff7ead7960261f2da2fc088987f3486e4460bc2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374026","number":374026,"commit":"1bcec8f3c2c40583fdcedc8ac44e73666839a584"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373011","number":373011,"commit":"f5f6e95525b64cae072fefcde80f80268432d45f"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372280","number":372280,"commit":"09ef49f420093871900d1ca8c104946c38ed3941"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372059","number":372059,"commit":"c8089ce5cdbc8a5054b29e05e2fd14d9fcb89f52"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372022","number":372022,"commit":"e78a797a2165c810527da3ed2d7cbcd07e033a34"}],"number":374090}
buildkite-pr-comment-->

kfirpeled reviewed (commented):
<img width="1017" height="138" alt="Image" src="https://github.com/user-attachments/assets/de9e342d-376e-4474-9f8b-11400e5a6942" />

Question: should we ask from the user to enable asset inventory?

kfirpeled reviewed (approved):
lgtm, added few comments
